sky/packages/sky/lib/src/widgets/scaffold.dart

```diff
@@ -6,247 +6,76 @@ import 'dart:sky' as sky;
-
-class _ScaffoldElement extends RenderObjectElement<Scaffold> {
-  _ScaffoldElement(Scaffold widget) : super(widget);
-
-  Map<_ScaffoldSlots, Element> _children;
-
-  _RenderScaffold get renderObject => super.renderObject;
-
-  void visitChildren(ElementVisitor visitor) {
-    for (_ScaffoldSlots slot in _ScaffoldSlots.values) {
-      Element element = _children[slot];
-      if (element != null)
-        visitor(element);
-    }
-  }
-
-  void mount(Element parent, dynamic newSlot) {
-    super.mount(parent, newSlot);
-    _children = new Map<_ScaffoldSlots, Element>();
-    for (_ScaffoldSlots slot in _ScaffoldSlots.values) {
-      Element newChild = widget._children[slot]?.createElement();
-      _children[slot] = newChild;
-      newChild?.mount(this, slot);
-    }
-  }
-
-  void update(Scaffold newWidget) {
-    super.update(newWidget);
-    assert(widget == newWidget);
-    for (_ScaffoldSlots slot in _ScaffoldSlots.values) {
-      _children[slot] = updateChild(_children[slot], widget._children[slot], slot);
-      assert((_children[slot] == null) == (widget._children[slot] == null));
-    }
-  }
-
-  void insertChildRenderObject(RenderObject child, _ScaffoldSlots slot) {
-    renderObject[slot] = child;
-  }
-
-  void moveChildRenderObject(RenderObject child, dynamic slot) {
-    removeChildRenderObject(child);
-    insertChildRenderObject(child, slot);
-  }
-
-  void removeChildRenderObject(RenderObject child) {
-    assert(renderObject == child.parent);
-    renderObject.remove(child);
+    this.body,
+    this.statusBar,
+    this.toolBar,
+    this.snackBar,
+    this.floatingActionButton
+  }) : super(key: key);
+
+  final Widget body;
+  final Widget statusBar;
+  final Widget toolBar;
+  final Widget snackBar;
+  final Widget floatingActionButton;
+
+  Widget build(BuildContext context) {
+    double toolBarHeight = 0.0;
+    if (toolBar != null)
+      toolBarHeight = kToolBarHeight + sky.view.paddingTop;
+
+    double statusBarHeight = 0.0;
+    if (statusBar != null)
+      statusBarHeight = kStatusBarHeight;
+
+    List<Widget> children = <Widget>[];
+
+    if (body != null) {
+      children.add(new Positioned(
+        top: toolBarHeight, right: 0.0, bottom: statusBarHeight, left: 0.0,
+        child: body
+      ));
+    }
+
+    if (statusBar != null) {
+      children.add(new Positioned(
+        right: 0.0, bottom: 0.0, left: 0.0,
+        child: new SizedBox(
+          height: statusBarHeight,
+          child: statusBar
+        )
+      ));
+    }
+
+    if (toolBar != null) {
+      children.add(new Positioned(
+        top: 0.0, right: 0.0, left: 0.0,
+        child: new SizedBox(
+          height: toolBarHeight,
+          child: toolBar
+        )
+      ));
+    }
+
+    if (snackBar != null) {
+      children.add(new Positioned(
+        right: 0.0, bottom: statusBarHeight, left: 0.0,
+        child: new Column([
+          new Padding(
+            padding: const EdgeDims.only(right: 16.0, bottom: 16.0),
```

Comment:
only include this if the floating action button is present, otherwise you'll have a hit test dead zone over the body whenever you have a snackbar